apps/src/blockly/addons/plusMinusBlocks/if.js

```diff
@@ -0,0 +1,253 @@
+      return null;
+    }
+    const container = GoogleBlockly.utils.xml.createElement('mutation');
+    container.setAttribute('elseif', this.elseIfCount_);
+    if (this.hasElse_) {
+      // Has to be stored as an int for backwards compat.
+      container.setAttribute('else', 1);
+    }
+    return container;
+  },
+
+  /**
+   * Parses XML to restore the else-if and else inputs.
+   * @param {!Element} xmlElement XML storage element.
+   * @this {GoogleBlockly.Block}
+   */
+  domToMutation: function (xmlElement) {
+    const targetCount = parseInt(xmlElement.getAttribute('elseif'), 10) || 0;
+    this.hasElse_ = !!parseInt(xmlElement.getAttribute('else'), 10) || 0;
+    if (this.hasElse_ && !this.getInput('ELSE')) {
+      this.appendStatementInput('ELSE')
+        .appendField(GoogleBlockly.Msg['CONTROLS_IF_MSG_ELSE'])
+        .appendField(createMinusField(), 'MINUS_ELSE');
+    }
+    this.updateShape_(targetCount);
+  },
+
+  /**
+   * Returns the state of this block as a JSON serializable object.
+   * @returns {{elseIfCount: (number|undefined),
+   *     haseElse: (boolean|undefined)}} The state of this block, ie the else
+   *     if count and else state.
+   */
+  saveExtraState: function () {
+    if (!this.elseIfCount_ && !this.hasElse_) {
+      return null;
+    }
+    const state = Object.create(null);
+    if (this.elseIfCount_) {
+      state['elseIfCount'] = this.elseIfCount_;
+    }
+    if (this.hasElse_) {
+      state['hasElse'] = true;
+    }
+    return state;
+  },
+
+  /**
+   * Applies the given state to this block.
+   * @param {*} state The state to apply to this block, ie the else if count and
+   *     else state.
+   */
+  loadExtraState: function (state) {
+    const targetCount = state['elseIfCount'] || 0;
+    this.hasElse_ = state['hasElse'] || false;
+    if (this.hasElse_ && !this.getInput('ELSE')) {
+      this.appendStatementInput('ELSE')
+        .appendField(GoogleBlockly.Msg['CONTROLS_IF_MSG_ELSE'])
+        .appendField(createMinusField(), 'MINUS_ELSE');
+    }
+    this.updateShape_(targetCount);
+  },
+
+  /**
+   * Adds else-if and do inputs to the block until the block matches the
+   * target else-if count.
+   * @param {number} targetCount The target number of else-if inputs.
+   * @this {GoogleBlockly.Block}
+   * @private
+   */
+  updateShape_: function (targetCount) {
+    while (this.elseIfCount_ < targetCount) {
+      this.addElseIf_();
+    }
+    while (this.elseIfCount_ > targetCount) {
+      this.removeElseIf_();
+    }
+  },
+
+  /**
+   * Callback for the plus field. Adds an else-if input to the block.
+   */
+  plus: function () {
+    if (this.hasElse_) {
+      this.addElseIf_();
+    } else {
+      this.addElse_();
+    }
+  },
+
+  /**
+   * Callback for the minus field. Triggers "removing" the input at the specific
+   * index.
+   * @see removeInput_
+   * @param {number} index The index of the else-if input to "remove". Undefined for else input.
```

Comment:
Based on comments below this `index` cannot be 0 right? Worth including that in the comment?

Reply:
Cannot be 0. Will document with a comment!